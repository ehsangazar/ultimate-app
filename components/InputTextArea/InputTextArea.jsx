/** @jsx jsx */
import { css, jsx } from '@emotion/core'
import { useTheme } from 'emotion-theming'

const InputTextArea = ({ onChange, placeholder }) => {
  const theme = useTheme()

  const handleOnChange = (event) => {
    onChange(event.target.value)
  }

  return (
    <div
      css={css`
        width: 100%;
        textarea::placeholder {
          direction: rtl;
        }
      `}
    >
      <textarea
        css={css`
          width: 100%;
          background-color: ${theme.colors.backgroundColorPrimary};
          font-size: ${theme.typography.paragraph};
          text-decoration: none;
          border: ${theme.border[1]} solid;
          border-color: ${theme.borderColor.light};
          padding: ${theme.spaces[4]};
          border-radius: ${theme.borderRadius[1]};
          display: inline-block;
          position: relative;
          min-height: 300px;
          cursor: pointer;
        `}
        autoComplete="new-password"
        onChange={handleOnChange}
        placeholder={placeholder}
      />
    </div>
  )
}

export default InputTextArea
