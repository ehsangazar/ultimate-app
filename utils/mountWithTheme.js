import React from 'react'
import { ThemeProvider } from 'emotion-theming'
import renderer from 'react-test-renderer'
import theme from '../configs/theme'

const mountWithTheme = (children) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  )
}

export default mountWithTheme
