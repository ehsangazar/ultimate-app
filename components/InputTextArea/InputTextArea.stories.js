import React from 'react'
import InputTextArea from './InputTextArea'

export default { title: 'InputTextArea' }

export const simple = () => <InputTextArea placeholder="توضیحات" />
