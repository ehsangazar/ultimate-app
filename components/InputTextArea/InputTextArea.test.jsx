import React from 'react'
import serializer from 'jest-emotion'
import mountWithTheme from '../../utils/mountWithTheme'
import InputTextArea from './InputTextArea'

expect.addSnapshotSerializer(serializer)

describe('InputTextArea', () => {
  test('Renders Correctly', () => {
    const component = mountWithTheme(<InputTextArea placeholder="sample" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
