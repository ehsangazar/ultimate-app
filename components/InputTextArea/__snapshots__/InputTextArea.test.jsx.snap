// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextArea Renders Correctly 1`] = `
.emotion-1 {
  width: 100%;
}

.emotion-1 textarea::-webkit-input-placeholder {
  direction: rtl;
}

.emotion-1 textarea::-moz-placeholder {
  direction: rtl;
}

.emotion-1 textarea:-ms-input-placeholder {
  direction: rtl;
}

.emotion-1 textarea::placeholder {
  direction: rtl;
}

.emotion-0 {
  width: 100%;
  background-color: #ecf0f1;
  font-size: font-size:1px;
  line-height: 1.6;
  font-weight: 300;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 2px solid;
  border-color: #bdc3c7;
  padding: 16px;
  border-radius: 4px;
  display: inline-block;
  position: relative;
  min-height: 300px;
  cursor: pointer;
}

<div
  className="emotion-1"
>
  <textarea
    autoComplete="new-password"
    className="emotion-0"
    onChange={[Function]}
    placeholder="sample"
  />
</div>
`;
